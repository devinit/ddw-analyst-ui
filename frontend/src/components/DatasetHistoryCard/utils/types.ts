export interface SavedQueryData {
  id: number;
  saved_query_db_table: string;
  operation: number;
  full_query: string | null;
  active: boolean;
  description: string;
  created_on: string;
  user: string;
  status: string;
}
