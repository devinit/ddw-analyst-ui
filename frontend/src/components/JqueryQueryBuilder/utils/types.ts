export type JqueryQueryBuilderFilter = {
  condition?: string;
  rules?: (JqueryQueryBuilderFilterComparator | JqueryQueryBuilderFilter)[];
};

export type JqueryQueryBuilderHaving = {
  condition?: string;
  rules?: (JqueryQueryBuilderHavingComparator | JqueryQueryBuilderHaving)[];
};

export type JqueryQueryBuilderHavingComparator = {
  id: string;
  field: string;
  type: string;
  input: string;
  operator: JqueryQueryBuilderComps;
  value?: string | number | { plain: string | number } | { column: string; aggregate: string };
  values?: { value: string; label: string }[];
};

export type JqueryQueryBuilderFilterComparator = {
  id: string;
  field: string;
  type: string;
  input: string;
  operator: JqueryQueryBuilderComps;
  value: string | number;
};
export type JqueryQueryBuilderComps =
  | 'equal'
  | 'not_equal'
  | 'less'
  | 'less_or_equal'
  | 'greater'
  | 'greater_or_equal'
  | 'between'
  | 'in';

export type JqueryQueryBuilderFieldData = {
  id: string | number;
  label: string;
  type: string;
};

export type JqueryQueryBuilderIcons = {
  add_group: string;
  add_rule: string;
  remove_group: string;
  remove_rule: string;
  error: string;
};
