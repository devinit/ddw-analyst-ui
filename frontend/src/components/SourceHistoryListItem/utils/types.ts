export interface FrozenData {
  id: number;
  parent_db_table: string;
  frozen_db_table: string | null;
  active: boolean;
  description: string;
  created_on: string;
  user: string;
  status: string;
  logs: string | null;
}
