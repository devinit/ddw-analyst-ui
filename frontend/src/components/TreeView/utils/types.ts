import { ReactElement } from 'react';

export interface TreeViewProps {
  data: Data[];
  depth?: number;

  deleteElement?: ReactElement;

  getStyleClassCb?: (node: Data) => void;

  isCheckable?: (node: Data, depth?: number) => void;
  isDeletable?: (node: Data, depth?: number) => void;
  isExpandable?: (node: Data, depth?: number) => void;

  keywordChildren?: string;
  keywordChildrenLoading?: string;
  keywordKey?: string;
  keywordLabel?: string;

  loadingElement?: ReactElement;
  noChildrenAvailableMessage?: string;

  onCheckToggleCb?: (data: Data[], depth?: number) => void;
  onDeleteCb?: (node: Data, data: Data[], depth?: number) => void;
  onExpandToggleCb?: (node: Data, depth?: number) => void;
  onUpdateCb?: (data: unknown[], depth?: number) => void;

  transitionEnterTimeout?: number;
  transitionExitTimeout?: number;
}

export interface Data {
  id: number;
  name: string;
  isExpanded?: boolean;
  isChecked?: boolean;
  isChildrenLoading?: boolean;
  children?: Data[];
}
