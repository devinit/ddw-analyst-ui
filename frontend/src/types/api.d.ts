export interface APIResponse<T> {
  count: number;
  next: string;
  previous: string;
  results: T;
}
export interface Links {
  next: string | null;
  previous: string | null;
}
export type LinksMap = Map<keyof Links, Links[keyof Links]>;

export interface FetchOptions {
  limit?: number;
  offset?: number;
  link?: string;
  search?: string;
  id?: number;
}
