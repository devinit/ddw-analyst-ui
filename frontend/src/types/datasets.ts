import { List, Map } from 'immutable';

export interface Dataset {
  id: string;
  title: string;
  description: string;
  publication: string;
  releasedAt: string;
  geography: string;
  geographicCode: string | null;
  createdBy: string;
  licence: string | null;
  citation: string | null;
}
export type DatasetMap = Map<keyof Dataset, Dataset[keyof Dataset]>;
export type DatasetList = List<DatasetMap>;
