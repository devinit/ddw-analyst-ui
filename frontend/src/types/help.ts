export type MenuLink = {
  caption: string;
  url: string;
  disabled?: boolean;
  addDividerAfter?: boolean;
};
