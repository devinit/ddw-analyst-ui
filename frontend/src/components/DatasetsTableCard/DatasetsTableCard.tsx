import * as React from 'react';
import { Card, FormControl } from 'react-bootstrap';
import { Dimmer, Loader } from 'semantic-ui-react';
import { Dataset, DatasetList } from '../../types/datasets';
import { DatasetTable } from '../DatasetTable';
import { DatasetTableRow } from '../DatasetTableRow';

interface ComponentProps {
  loading: boolean;
  datasets: DatasetList;
}

export const DatasetsTableCard: React.SFC<ComponentProps> = props => {
  const onRowClick = () => {
    console.log('Clicked'); //tslint:disable-line
  };
  const renderTableRows = () => {
    if (props.datasets && props.datasets.size) {
      return props.datasets.map((dataset, index) =>
        <DatasetTableRow
          key={ index }
          onClick={ onRowClick }
          dataset={ dataset.toJS() as Dataset }
        >
          <span>Actions Go Here!</span>
        </DatasetTableRow>
      );
    }

    return null;
  };

  return (
    <React.Fragment>
      <Dimmer active={ props.loading } inverted>
        <Loader content="Loading..." />
      </Dimmer>
      <Card>
        <Card.Header className="card-header-text card-header-danger">
          <Card.Title>
            <FormControl
              placeholder="Search ..."
              className="w-50"
              data-testid="dataset-table-search"
            />
          </Card.Title>
        </Card.Header>
        <Card.Body>
          <DatasetTable>{ renderTableRows() }</DatasetTable>
        </Card.Body>
      </Card>
    </React.Fragment>
  );
};

export { DatasetsTableCard as default };
