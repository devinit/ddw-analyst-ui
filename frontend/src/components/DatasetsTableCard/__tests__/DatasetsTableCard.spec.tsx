/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render } from '@testing-library/react';
import { List, fromJS } from 'immutable';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { DatasetMap } from '../../../types/datasets';
import { DatasetsTableCard } from '../DatasetsTableCard';

const datasets: DatasetMap[] = [
  fromJS({
    id: '1',
    title: 'Dataset 1',
    publication: 'My first dataset',
    releasedAt: new Date()
  }),
  fromJS({
    id: '2',
    title: 'Dataset 2',
    publication: 'My second dataset',
    releasedAt: new Date()
  })
];

test('renders correctly with no datasets', () => {
  const { getByTestId } = render(<DatasetsTableCard loading={ false }/>);

  expect(getByTestId('datasets-no-data')).toHaveTextContent('No results found');
});

test('renders the datasets correctly', () => {
  const renderer = TestRenderer
    .create(<DatasetsTableCard loading={ false } datasets={ List(datasets) }/>)
    .toJSON();

  expect(renderer).toMatchSnapshot();
});

test('clicking the details button on a row opens the modal', () => {
  const { getByTestId, queryByTestId } = render(<DatasetsTableCard loading={ false } datasets={ List(datasets) }/>);

  expect(queryByTestId('details-modal-1')).toBeFalsy();

  fireEvent.click(getByTestId('details-button-1'));

  expect(queryByTestId('details-modal-1')).toBeTruthy();
  expect(getByTestId('details-modal-1').parentElement).toHaveClass('show');
});
