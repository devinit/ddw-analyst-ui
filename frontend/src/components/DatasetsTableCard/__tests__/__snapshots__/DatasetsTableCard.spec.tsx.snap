// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the default component correctly 1`] = `
Array [
  <div
    className="ui inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="ui text loader"
      >
        Loading...
      </div>
    </div>
  </div>,
  <div
    className="card"
  >
    <div
      className="card-header-text card-header-danger card-header"
    >
      <div
        className="card-title h5"
      >
        <input
          className="w-50 form-control"
          data-testid="dataset-table-search"
          placeholder="Search ..."
        />
      </div>
    </div>
    <div
      className="card-body"
    >
      <div
        className="table-responsive"
      >
        <table
          className="table sources-table table-striped table-hover"
        >
          <thead>
            <tr>
              <th>
                Title
              </th>
              <th>
                Publication
              </th>
              <th>
                Release Date
              </th>
              <th
                className="text-right"
              >
                Actions
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              data-testid="datasets-table-row"
              onClick={[Function]}
            >
              <td>
                Dataset 1
              </td>
              <td>
                My first dataset
              </td>
              <td>
                Mon Oct 07 2019
              </td>
              <td
                className="text-right"
              >
                <span>
                  Actions Go Here!
                </span>
              </td>
            </tr>
            <tr
              data-testid="datasets-table-row"
              onClick={[Function]}
            >
              <td>
                Dataset 2
              </td>
              <td>
                My second dataset
              </td>
              <td>
                Mon Oct 07 2019
              </td>
              <td
                className="text-right"
              >
                <span>
                  Actions Go Here!
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
]
`;
